pkg/cmd/roachtest/roachtestutil/disconnect.go

```diff
@@ -0,0 +1,58 @@
+// Copyright 2023 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package roachtestutil
+
+import (
+	"context"
+
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster"
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/test"
+	"github.com/stretchr/testify/require"
+)
+
+// Disconnect takes a set of nodes and disconnects each node from all others in
+// the list. This function requires root privileges.
+//
+// TODO (msbutler):  consider a different api: Disconnect(t test.Test, c
+// cluster.Cluster, ctx context.Context, node int, toDisconnect []int) where we
+// disconnect a node from all nodes in toDisconnect. Adding iptable rules on
+// both sides shouldn't be necessary.
+func Disconnect(ctx context.Context, t test.Test, c cluster.Cluster, nodes []int) {
```

Comment:
I would also argue for a less generic name than `Disconnect`. Would likely be good to scope this `Disconnect + Cleanup` to a struct instead of package-level functions.

Reply:
Something else to consider: as an API, I'd much prefer this function to return an error and let the caller decide if the test should fail and how the error should be reported.

I understand this was a quick add to support a c2c roachtest, but consider if and how this could be used in other roachtests too: `failover/*` roachtests, for example, also do a fair bit of `iptables` manipulation. Reuse is generally good.

Reply:
yeah, this all makes sense! I'll address these in https://github.com/cockroachdb/cockroach/issues/103887

Reply:
this is dumb, I'll get rid of my crufty API in the morning in this PR.

Reply:
@renatolabs I refactored this PR to directly use the `blackholeFailer` struct directly instead. I debated creating a new interface `failerWithSetupCluster`, but that seemed like more trouble than it was worth.

Reply:
LGTM!